src/App.tsx

```diff
@@ -21,14 +73,18 @@ export const App: React.FC = () => {
             </div>
 
             <div className="block">
-              <Loader />
-              <TodoList />
+              {currentTodo && <TodoModal todo={currentTodo} />}
+              {showLoader && <Loader />}
+              {visibleTodos.length === 0 && query.length > 0 && (
+                <p className="notification is-warning">
+                  There are no todos matching current filter criteria
+                </p>
+              )}
+              {visibleTodos.length > 0 && <TodoList todoList={visibleTodos} />}
```

Comment:
Let's add some spaces between elements to improve readability

And create a variable for this conditional `visibleTodos.length === 0 && query.length > 0`

Reply:
done